platform-sdk/docs/proposals/DS-0x/DS - Tss block signing - Cryptography APIs.md

```diff
@@ -0,0 +1,458 @@
+- **Groth 21**: A zero-knowledge proof system developed by Jens Groth in 2021. It is a non-interactive zero-knowledge (zk-SNARK) protocol, which allows one party (the prover) to convince another party (the verifier) that a statement is true without revealing any information beyond the validity of the statement itself.
+- **ElGamal**: In 1985 ElGamal proposed a randomized signature scheme called ElGamal signature scheme. On a message and a random number chosen by a signer, ElGamal signature scheme produces a signature consisting of two numbers (r, s), where r is computed from a random number regardless of a message and then s is computed from a message, a signer's secret key, a chosen random number, and r.
+- **EC ([Elliptic-Curves](https://crypto.stanford.edu/pbc/notes/elliptic))**: "Elliptic" is not elliptic in the sense of an "oval circle". In the field "Fp", an EC is like a non-connected cloud of points. All points satisfy an equation and all operations are performed modulo p. Some elliptic curves are pairing-friendly.
+- **Bilinear Pairings**: Are mathematical functions used in cryptography to map two elements of different groups (in EC, the group is an elliptic curve) to a single value in another group in a way that preserves certain algebraic properties.
+- **Fields**: Mathematical structures where addition, subtraction, multiplication, and division are defined and behave as expected (excluding division by zero).
+- **Groups**: Sets equipped with an operation (like addition or multiplication) that satisfies certain conditions (closure, associativity, identity element, and inverses).
+- **Share**: Represents a piece of the public/private necessary elements to create or verify a signature. In TSS, a threshold number of shares is needed to validate a signature. To represent weight, any participant can have more than 1 share.
+- **Polynomial Commitment**: A polynomial commitment is a cryptographic commitment to a polynomial. It enables a committer to commit to a polynomial such that: a)The committer can later reveal the polynomial and prove that it was indeed the committed polynomial and b)The commitment allows verifying evaluations of the polynomial at specific points without revealing the entire polynomial.
+
+### Goals
+- **Usability:** Design a user-friendly library that is easy to integrate with consensus-node code and block-node code.
+- **EVM support:** Our implementation must be supported by EVM.
+- **Security:** Our APIS should use audited code and be able to pass security audits both internal and external.
+- **Flexibility**: Minimize the impact of introducing support for other elliptic curves. 
+- **Independent Release:** Design release cycle independent components that can be used both in the consensus node and in the future block node.
+
+### Non-Goals
+- Build our own cryptography library implementation that supports EC-curves from the gecko in Java.
```

Comment:
I added it and then took it back; what would be your preferred wording for this point? 

It cannot be that it will not be publicly available or accessible because it will be open source and publicly available. Do we care if there are external users of this library? It is not the primary design goal, but neither is it for metrics or config API...